### PR TITLE
docs(#92): require positive, negative, and edge-case test coverage in agents.md

### PR DESCRIPTION
**Pull request by RudoiDmytro**
## Summary

Strengthens the root `agents.md` so AI agents can no longer treat a single
happy-path test as adequate coverage. Adds a clearly labeled, mandatory
**Test-Scenario Coverage Policy** and wires it into the existing agent workflows.

Closes #92.

## What changed (`agents.md` only — docs)

- **New `## Mandatory Test-Scenario Coverage Policy` section** (before
  `## Testing Strategies`), framed as a hard requirement with five ordered steps:
  1. **Pick the right test layer** — a table mapping change → suite → command
     (client unit, server unit, integration, E2E, visual) plus specialized suites
     (mutation, memory-leak, load, Lighthouse).
  2. **Cover every applicable scenario class** — positive / happy, negative /
     invalid / failure, and boundary / edge — backed by an explicit 9-item checklist.
  3. **Document any skipped scenario class** with a concrete `Not applicable: <reason>`
     (bare "not applicable" or silent omission is rejected).
  4. **Regression coverage is mandatory for bug fixes** — a test that fails before
     the fix and passes after, unless a concrete recorded reason prevents it.
  5. **Verify before calling test work done** — run the relevant `make` targets.
- **Cross-references** so the policy governs every workflow agents follow: Adding a
  Feature, Fixing Bugs, Testing Strategies, After Making Changes, Module Checklist,
  and the Code Review Self-Check.
- **Reframed the `>80%` coverage line** as a floor that never substitutes for
  scenario-class coverage (closing a weaker, citable standard).

## Acceptance criteria mapping

- **AC1** clearly labeled mandatory policy → new top-level section.
- **AC2** positive/negative/edge explicitly required → Step 2 + 9-item checklist.
- **AC3** concrete justification for skipped classes → Step 3.
- **AC4** points to repo verification commands → Step 1 table + Step 5 gate (all
  `make` targets verified against the `Makefile`).
- **AC5** regression mandatory for bug fixes → Step 4 + Fixing Bugs cross-ref.

All nine "specific gaps to close" are covered by the Step 2 checklist: valid /
invalid input, empty/null/undefined/partial data, loading/timeout/retry/error,
permission/auth/role, boundary/off-by-one, locale/i18n, visual regressions, and
previously-fixed-bug regressions.

## Verification

- `make lint-md` → pass (exit 0) — the exact command CI's `static testing`
  workflow runs via `make lint`.
- All new/edited lines ≤ 100 chars (markdownlint MD013); no duplicate headings;
  allowed fence languages; Prettier clean (pre-commit hook).
- Docs-only change — no source/UI/HTML/CSS touched, so accessibility review is
  not applicable.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---
## Summary by cubic
Adds a Mandatory Test-Scenario Coverage Policy in `agents.md` (positive, negative, edge) with the "Not applicable: <reason>" convention, and wires it into workflows. Streamlines auth by replacing `@apollo/client` reactive var, deferring DI until first action, adopting the `@auth` alias, and raises the mobile Lighthouse gate to 0.85. Addresses #92.

- **New Features**
  - 5-step policy: pick test layer(s) → cover positive/negative/edge via checklist → justify any "Not applicable: <reason>" → require regression tests for fixes → verify with `make` targets.
  - Wired into feature/bug workflows and testing docs; `>80%` is a floor. Added `make` commands for visual/E2E/unit/integration/lint/format.

- **Refactors**
  - Deferred auth actions lazily load the DI graph on first use; set loading flags before await; log and surface a neutral, retryable failure; retry after failure (no rejection caching).
  - Replaced `@apollo/client` reactive var with a lightweight `ReactiveVarFactory` + `useSyncExternalStore` (skips same-ref writes, isolates throwing listeners); added `reactive-var` types; adopted the `@auth` alias; updated `use-auth-token` path and `ProtectedRoute`.
  - Raised mobile Lighthouse threshold to 0.85 and updated its unit test; added unit/integration tests for deferred auth actions and the reactive var, including the throwing-listener path.

<sup>Written for commit 8b7aab5251f27c667314c1107b17ca611ba36c63. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/VilnaCRM-Org/crm/pull/99?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: agents.md [hunk 1]
```diff
@@ -650,8 +653,96 @@ export const UIComponentName: React.FC<UIComponentNameProps> = (props) => {
...
+- [ ] Previously fixed bug regression coverage (see Step 4)
+
+### Step 3 — Document Any Skipped Scenario Class
+
+If a scenario class or checklist item genuinely does not apply, record it explicitly with a
+concrete reason — in the test file (as a comment), the PR description, or your task summary.
+Use the same `Not applicable: <reason>` convention as the Mandatory Skill Check. A bare
+"not applicable" with no reason, or silent omission, does not satisfy this policy.
+
+Example: `Boundary/edge: N/A — presentational component with no inputs, state, or branches.`
```
[the remark this answers is not in the document]
Comment (RudoiDmytro) [reply agents.md]: Fixed in 8b7aab5: the example now uses the exact `Not applicable: <reason>` convention — `Boundary/edge — Not applicable: presentational component with no inputs, state, or branches.`

**Review by Kravalg — APPROVED** [review]
